File: LeetCode/PytonProblems/LC-1-TwoSum.py
# Given an array of integers nums and an integer target,
# return indices of the two numbers such that they add up to target.
# You may assume that each input would have exactly one solution, and you may not use the same element twice.
# You can return the answer in any order.

class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        numMap = {}
        length = len(nums)

        for i in range(length):
            complement = target - nums[i]
            if complement in numMap:
                return[i, numMap[complement]]
            numMap[nums[i]] = i

File: LeetCode/JavaProblems/LC-1-TwoSum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap<Integer,Integer> numMap = new HashMap<Integer,Integer>();

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (numMap.containsKey(complement)) {
                return new int[]{i,numMap.get(complement)};
            }
            numMap.put(nums[i], i);
        }
        return new int[]{};
    }
}